### PR TITLE
Support latest rxdart version and align Dart SDK version

### PR DESCRIPTION
**Pull request by Carapacik**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Update versions and align sdk

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/Baseflow/flutter_cache_manager/blob/develop/CONTRIBUTING.md))
- [x] Relevant documentation was updated
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by Carapacik** [comment]
@mvanbeusekom check plz

**Comment by Carapacik** [comment]
@mvanbeusekom @renefloor check this changes

**Review by hoc081098 — APPROVED** [review]
:shipit:

**Comment by hoc081098** [comment]
Hi maintainers, please publish it 🙏

File: flutter_cache_manager/pubspec.yaml [hunk 1]
```diff
@@ -17,13 +17,13 @@ dependencies:
   http: ^1.2.1
   path: ^1.9.0
   path_provider: ^2.1.3
-  rxdart: ^0.27.7
+  rxdart: '>=0.27.7 <1.0.0'
```
Comment (renefloor) [inline flutter_cache_manager/pubspec.yaml:20]: We have to guarantee to be compatible with other breaking changes, we should only allow 0.27 and 0.28:
```dart
  rxdart: '>=0.27.7 <0.29.0'
```

File: flutter_cache_manager/example/pubspec.yaml [hunk 2]
```diff
@@ -3,21 +3,21 @@ description: A project that showcases usage of flutter_cache_manager
...
 dependencies:
   baseflow_plugin_template: ^2.1.2
-  cupertino_icons: ^1.0.5
+  cupertino_icons: ^1.0.8
   flutter:
     sdk: flutter
   flutter_cache_manager:
     path: ../
-  url_launcher: ^6.1.10
+  url_launcher: ^6.3.0
```
Comment (renefloor) [inline flutter_cache_manager/example/pubspec.yaml:15]: I would prefer not to have all these changes where you update a minor or patch version. That only limits users of this library to update when they might not want that. For example some user might have a reason to stick to url_launcher 6.2.0 and that should work fine with this library. I always try to keep the support as wide as possible and only limit when it gives big benefits..

Comment (Carapacik) [reply flutter_cache_manager/example/pubspec.yaml:15]: Ok

File: flutter_cache_manager/example/pubspec.yaml [hunk 3]
```diff
@@ -3,21 +3,21 @@ description: A project that showcases usage of flutter_cache_manager
 publish_to: none
 version: 1.0.0+1
 environment:
-  sdk: '>=2.17.0 <4.0.0'
+  sdk: '>=3.0.0 <4.0.0'
```
[the remark this answers is not in the document]
Comment (Carapacik) [reply flutter_cache_manager/example/pubspec.yaml:6]: Other dependencies support versions higher than 3.0

File: flutter_cache_manager/pubspec.yaml (hunk 1 above)
Comment (Carapacik) [reply flutter_cache_manager/pubspec.yaml]: ok

File: flutter_cache_manager/example/pubspec.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Carapacik) [reply flutter_cache_manager/example/pubspec.yaml:6]: For example, http and file have the minimum version of dart 3.0

**Review by hoc081098 — APPROVED** [review]
Hello maintainers, please merge this PR :shipit: .
Thank you so much :pray:

**Comment by renefloor** [comment]
@Carapacik I hope you don't mind that I pushed a commit to your branch. I want to set the version of flutter_lints, so everybody has the same warnings. Also fixed 1 warning and some other versioning stuff.

**Pull request merged**